include/envoy/network/connection.h

```diff
@@ -120,6 +120,7 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
```

Comment:
IpList::contains(const Address::Instance& address) calls CidrRange::isInRange(InstanceConstSharedPtr address). 
This is trying to solve the problem of the Instance->string->InstanceConstSharedPtr conversion when calling isInRange. The way this is done is to change IpList::contains to take an InstanceConstSharedPtr instead.